variables.tf

```
@@ -0,0 +1,37 @@
+variable "gitlab_url" {
+  type        = string
+  default     = "gitlab.com"
+  description = "GitLab URL. The address of your GitLab instance, such as gitlab.com or gitlab.example.com."
+
+  validation {
+    condition     = !can(regex("^(http|tls)", var.gitlab_url))
+    error_message = "URL should not start with any prefix like http(s) or tls"
+  }
+}
```

```suggestion
variable "gitlab_host" {
  type        = string
  default     = "gitlab.com"
  description = "GitLab host, such as gitlab.com or gitlab.example.com"

  validation {
    condition = can(regex("^(?!-)[A-Za-z0-9-]+(\\.[A-Za-z0-9-]+)*$", var.hostname))
    error_message = "Please provide a valid hostname format (e.g. gitlab.com or gitlab.example.com)"
  }
}
```

I think `gitlab_host` is more application here, having URL makes people want to supply with the protocol but we don't use that anywhere.